src/TenorGuitar/Types.purs

```diff
@@ -0,0 +1,60 @@
+module TenorGuitar.Types where
+
+import Prelude (negate)
+import Data.Maybe (Maybe)
+
+-- | a finger position on a string
+-- | n < 0  : String is silent
+-- | n == 0 : Open string is sounded
+-- | N > 0  : string is fretted at this position
+type FingerPosition = Int
+
+-- | fingering is represented as an array of finger positions,
+-- | one for each of the 4 strings
+type Fingering = Array FingerPosition
+
+-- | a fingered string
+type FingeredString =
+  { stringNumber :: Int
+  , fretNumber :: FingerPosition
+  }
+
+-- | a barreé
+type Barre =
+  Maybe FingeredString
+
+-- | The editable chord shape of the tenor guitar
+type ChordShape =
+  { name :: String -- the chord name
+  , firstFretOffset :: Int -- which fret on the tenor guitar does fret 1 represent
+  , barre :: Barre -- a barré at this fret from this string to
+  -- the final string (4)
+  , fingering :: Fingering -- the fingering
+  }
+
+data MouseAction
+  = Barre FingeredString
+  | OneFret FingeredString
+  | NoFret
+
+-- | the number of frets we display
+displayedFretCount :: Int
+displayedFretCount = 4
```

Comment:
Is 4 frets enough? See, for example http://chords.cc/en/tenorguitar/cgda/c/major. I bow to your superior knowledge here!

Reply:
I added the same amount of frets to display as the guitar (i.e. 6). I think that is a reasonable amount:

![image](https://user-images.githubusercontent.com/47760695/194781001-d5529d72-f1ea-44dd-86f9-0c2f9f826233.png)